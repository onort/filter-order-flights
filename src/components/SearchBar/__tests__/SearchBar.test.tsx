/* tslint:disable jsx-key */
import React from "react"
import { shallow, mount } from "enzyme"

import {
  Arrival,
  ArrivalDate,
  Departure,
  DepartureDate,
  Person
} from "../../../assets/icons"
import { Button, FormField, SearchBar } from "../../"

describe("<SearchBar />", () => {
  it("should match snapshot", () => {
    const wrapper = shallow(<SearchBar />)
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.html()).toMatchSnapshot()
  })

  it("should have correct children", () => {
    const wrapper = mount(<SearchBar />)
    expect(
      wrapper.containsAllMatchingElements([
        <FormField icon={Departure} size="normal" />,
        <FormField icon={Arrival} size="normal" />,
        <FormField icon={DepartureDate} size="normal" />,
        <FormField icon={ArrivalDate} size="normal" />,
        <FormField icon={Person} size="small" />,
        <Button text="Yeniden Ara" primary={true} />
      ])
    )
    wrapper.unmount()
  })

  it("should have a custom classname", () => {
    const testClassName = "testClassName"
    const wrapper = shallow(<SearchBar className={testClassName} />)
    expect(wrapper.find("div.container").hasClass(testClassName)).toBe(true)
  })
})
