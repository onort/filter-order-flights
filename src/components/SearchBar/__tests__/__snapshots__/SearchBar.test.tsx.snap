// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchBar /> should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchBar />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <FormField
          icon="departure.svg"
          size="normal"
        />,
        <FormField
          icon="arrival.svg"
          size="normal"
        />,
        <FormField
          icon="deparutre-date.svg"
          size="normal"
        />,
        <FormField
          icon="arrival-date.svg"
          size="normal"
        />,
        <FormField
          icon="person.svg"
          size="small"
        />,
        <Button
          className="searchButton"
          primary={true}
          text="Yeniden Ara"
          type="button"
        />,
      ],
      "className": "container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "icon": "departure.svg",
          "size": "normal",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "icon": "arrival.svg",
          "size": "normal",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "icon": "deparutre-date.svg",
          "size": "normal",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "icon": "arrival-date.svg",
          "size": "normal",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "icon": "person.svg",
          "size": "small",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "className": "searchButton",
          "primary": true,
          "text": "Yeniden Ara",
          "type": "button",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <FormField
            icon="departure.svg"
            size="normal"
          />,
          <FormField
            icon="arrival.svg"
            size="normal"
          />,
          <FormField
            icon="deparutre-date.svg"
            size="normal"
          />,
          <FormField
            icon="arrival-date.svg"
            size="normal"
          />,
          <FormField
            icon="person.svg"
            size="small"
          />,
          <Button
            className="searchButton"
            primary={true}
            text="Yeniden Ara"
            type="button"
          />,
        ],
        "className": "container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "icon": "departure.svg",
            "size": "normal",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "icon": "arrival.svg",
            "size": "normal",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "icon": "deparutre-date.svg",
            "size": "normal",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "icon": "arrival-date.svg",
            "size": "normal",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "icon": "person.svg",
            "size": "small",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "className": "searchButton",
            "primary": true,
            "text": "Yeniden Ara",
            "type": "button",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;

exports[`<SearchBar /> should match snapshot 2`] = `"<div class=\\"container\\"><div class=\\"container\\"><img class=\\"icon\\" src=\\"departure.svg\\"/><input class=\\"input\\"/></div><div class=\\"container\\"><img class=\\"icon\\" src=\\"arrival.svg\\"/><input class=\\"input\\"/></div><div class=\\"container\\"><img class=\\"icon\\" src=\\"deparutre-date.svg\\"/><input class=\\"input\\"/></div><div class=\\"container\\"><img class=\\"icon\\" src=\\"arrival-date.svg\\"/><input class=\\"input\\"/></div><div class=\\"container small\\"><img class=\\"icon\\" src=\\"person.svg\\"/><input class=\\"input\\"/></div><button class=\\"button primary searchButton\\" type=\\"button\\"><span class=\\"text\\">Yeniden Ara</span></button></div>"`;
